### PR TITLE
[SPARK-25392][CORE][WEBUI] Prevent error page when accessing pools page from history server

### PR DESCRIPTION
**Pull request by shahidki31**
### Why are the changes needed?

Currently from history server, we will not able to access the pool info, as we aren't writing pool information to the event log other than pool name. Already spark is hiding pool table when accessing from history server. But from the pool column in the stage table will redirect to the pools table, and that will throw error when accessing the pools page. To prevent error page, we need to hide the pool column also in the stage table

### Does this PR introduce any user-facing change?

No

### How was this patch tested?
Manual test

Before change:
Screenshot 2019-11-21 at 6 49 40 AM
Screenshot 2019-11-21 at 6 48 51 AM

After change:
Screenshot 2019-11-21 at 7 29 01 AM

### CODE REVIEW

**Comment by shahidki** [comment]
cc @srowen @vanzin Kindly review. https://github.com/apache/spark/pull/22571 tried to fix the issue by writing to the event log. But that isn't fully correct as the running tasks in the pool info will be incorrect. Already spark UI is hiding the pool table for history server, so this change seems more reasonable and it will prevent error in the UI.

**Comment by shahidki** [comment]
Jenkins, test this please

**Comment by shahidki** [comment]
retest this please

**Comment by shahidki** [comment]
Jenkins, retest this please

**Comment by shahidki31** [comment]
cc @vanzin @srowen Could you please review this patch. seems a minor change to fix avoiding the error page.

**Comment by shahidki31** [comment]
Jenkins, retest this please

**Comment by vanzin** [comment]
There are other places in the code where I see a check for `isFairScheduler` that is not affected by your change (e.g. where `AllStagesPage` instantiates a `StageTableBase`).

Shouldn't all places that check for `isFairScheduler` do it consistently to avoid this problem?

**Comment by shahidki31** [comment]
@vanzin Yes, right. I intended to change that too. That is why I changed the method `isFairScheduler`. Seems didn't push the change.

**Comment by dongjoon-hyun** [comment]
ok to test

File: core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala [hunk 1]
```diff
@@ -30,7 +30,8 @@ import org.apache.spark.ui.{UIUtils, WebUIPage}
 private[ui] class AllStagesPage(parent: StagesTab) extends WebUIPage("") {
   private val sc = parent.sc
   private val subPath = "stages"
-  private def isFairScheduler = parent.isFairScheduler
+  // Show pool information for only live UI
+  private def isFairScheduler = sc.isDefined && parent.isFairScheduler
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala]: It would be better to fix `StageTab.isFairScheduler` and remove this method.

Comment (shahidki31) [reply core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala]: Done

**Comment by vanzin** [comment]
Merging to master / 2.4.

**Pull request closed without merging**